src/3d/qgs3dmapscene.cpp

```diff
@@ -249,11 +250,13 @@ Qgs3DMapScene::Qgs3DMapScene( Qgs3DMapSettings &map, QgsAbstract3DEngine *engine
 
 void Qgs3DMapScene::viewZoomFull()
 {
-  QgsRectangle extent = sceneExtent();
-  float side = std::max( extent.width(), extent.height() );
-  float a =  side / 2.0f / std::sin( qDegreesToRadians( cameraController()->camera()->fieldOfView() ) / 2.0f );
-  // Note: the 1.5 multiplication is to move the view upwards to look better
-  mCameraController->resetView( 1.5 * std::sqrt( a * a - side * side ) );  // assuming FOV being 45 degrees
+  const QgsDoubleRange yRange = elevationRange();
+  const QgsRectangle extent = sceneExtent();
+  const double side = std::max( extent.width(), extent.height() );
+  double d = side / 2 / std::tan( qDegreesToRadians( cameraController()->camera()->fieldOfView() / 2 ) );
```

Comment:
I'd rather avoid the unnecessary qt macro here and just use standard c++ for degrees to radians

Reply:
done